Sources/Build/BuildPlan.swift

```diff
@@ -657,24 +683,45 @@ public final class SwiftTargetBuildDescription {
         self.tempsPath = buildParameters.buildPath.appending(component: target.c99name + ".build")
         self.derivedSources = Sources(paths: [], root: tempsPath.appending(component: "DerivedSources"))
         self.pluginDerivedSources = Sources(paths: [], root: buildParameters.dataPath)
+        self.pluginDerivedResources = []
         self.buildToolPluginInvocationResults = buildToolPluginInvocationResults
         self.prebuildCommandResults = prebuildCommandResults
 
+        func handlePluginDerivedFile(path absPath: AbsolutePath) {
+            // FIXME: Carry `additionalFileRules` until here.
+            let rule = TargetSourcesBuilder.computeRule(for: absPath, toolsVersion: toolsVersion, additionalFileRules: [])
+            print("Determined \(rule) for generated file at \(absPath)")
+            switch rule {
+            case .compile:
+                if absPath.extension == "swift" {
+                    let relPath = absPath.relative(to: self.pluginDerivedSources.root)
+                    self.pluginDerivedSources.relativePaths.append(relPath)
+                } else {
+                    // FIXME: Error for unsupported source file type.
+                }
+            case .copy, .processResource:
+                if let resource = TargetSourcesBuilder.resource(for: absPath, with: rule) {
+                    self.pluginDerivedResources.append(resource)
+                } else {
+                    // FIXME: Error for not obtaining a `Resource` object for a file.
+                }
+            case .header, .ignored, .modulemap, .none:
+                break // Ignored for now.
```

Comment:
should this throw / warn?

Reply:
Probably warn, I'd say.  That's what we do for other unhandled files.

Reply:
Don't have a strong opinion, but I think error seems appropriate since a plugin is essentially broken/incomplete if it generates files that aren't handled at all. I can see the argument that it should be a warning because the client of a plugin should maybe be able to use a broken plugin anyway?

Reply:
Yeah, that's a good point, it's different from just having a file that I added to my sources but I don't want to be blocked from building because of.  I don't feel strongly about this though.  I think it would be fine to start with a warning.

Reply:
Switching to a warning for now.